packages/lib-core/src/types/common.ts

```diff
@@ -0,0 +1,42 @@
+import type { AnyObject } from '@monorepo/common';
+import type { ReactNode } from 'react';
+
+// Type for extension hook
+export type ExtensionHook<TType, TOptions extends AnyObject = AnyObject> = (
+  options: TOptions,
+) => [TType, boolean, unknown];
```

Comment:
TypeScript 4+ allows tuple elements to be named, we already do that in tuple type `UseResolvedExtensionsResult`. We should improve this in a follow-up change.

Reply:
Yeah, that makes sense. I'll keep as is, and we can iterate on it.